app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt

```diff
@@ -24,7 +26,7 @@ class AboutEventViewModel(private val eventService: EventService) : ViewModel()
 
     fun loadEvent(id: Long) {
         if (id.equals(-1)) {
-            mutableError.value = "Error fetching event"
+            mutableError.value = OpenEventGeneral.appResource.getString(R.string.error_fetching_event_message)
```

Comment:
This is still using resources in viewmodel

Easy way to check if you are using resources in ViewModel - Instantiate the ViewModel in a test and run all methods

If the test crashes, you are using Android class in ViewModel

Reply:
I understand that now and referencing string resource inside ViewModel is not safe, so I don't find a way to extract those strings inside ViewModel.

And I have read the Android documentation, and it said "if the ViewModel needs the Application context, it can extend the AndroidViewModel class and have a constructor that receives the Application in the constructor". Should I use this technique to extract the hard-coded strings inside the view model into res/values/strings file or I should leave them unextracted?

Reply:
https://github.com/fossasia/open-event-android/blob/development/app/src/main/java/org/fossasia/openevent/general/data/Resource.kt